DOCUMENTATION.md

```diff
@@ -107,19 +122,162 @@ In case it's a **dictionary** with fields, please make sure the **encoder** func
+
+
+## ApplicationClient
+The class constructor can be called following this scheme:
+```python
+ApplicationClient(app_id, access_key_or_token, [net_address], [certificate], [discovery_address])
+```
+- `app_id`: **string**  this the name given to the application when it was created.
+![Screenshot of the console with app section](./images/app-console.png?raw=true)
+- `access_key_or_token`: **string**  this can be found at the bottom of the application page under **ACCESS KEYS**. You will need a key allowing you to change the settings if you wish to update your application.
+- `net_address`: **string**  this  is the address of the handler to which the application was registered. It needs to be provided as a `net_address=value` argument when calling the constructor.
+- `certificate`: **string**  this is the content of the certificate used to connect in a secure way to the handler. It needs to be provided as a `certificate=value` argument when calling the constructor.
+- `discovery_address`: **string**  this is the address of the discovery server to use in order to find back the address of the handler to which the application in registered. It needs to be provided as a `discovery_address=value` argument when calling the constructor.
+- `path_to_key=`: **string**  this is the absolute path to the file which contains the key from which the token you wish to use, is signed. It needs to be provided as a `path_to_key=value` argument when calling the constructor.
+The constructor returns an **ApplicationClient** object set up with the application informations, ready to get the application registered on The Things Network.
+
+### get
+Gives back the [**Application**](#application) object with the id given to the constructor.
+```python
+client.get()
+```
+
+### set_payload_format
+Sets the payload format of the application.
+```python
+client.set_payload_format(payload_format)
+```
+- payload_format: **string**  the new payload format.
+
+### set_custom_payload_functions
+Sets the payload functions of the application.
+```python
+client.set_custom_payload_functions([decoder], [encoder], [validator], [converter])
+```
+- `decoder`: **string**  decoder function that must be written in javascript, it needs to be provided as a `decoder=value` argument when calling the method.
+- `encoder`: **string**  encoder function that must be written in javascript, it needs to be provided as a `encoder=value` argument when calling the method.
+- `validator`: **string**  validator function that must be written in javascript, it needs to be provided as a `validator=value` argument when calling the method.
+- `converter`: **string**  converter function that must be written in javascript, it needs to be provided as a `converter=value` argument when calling the method.
+
+### set_register_on_join_access_key
+Sets the register on join access key of the application.
```

Comment:
Can you add more info in regards to what this does? https://github.com/TheThingsNetwork/api/blob/master/handler/handler.pb.go#L202